ambry-api/src/main/java/com.github.ambry/account/Account.java

```diff
@@ -0,0 +1,263 @@
+        this.status = metadata.getString(ACCOUNT_STATUS_KEY);
+        this.metadata = metadata;
+        JSONArray containerArray = metadata.getJSONArray(CONTAINERS_KEY);
+        for (int index = 0; index < containerArray.length(); index++) {
+          JSONObject containerMetadata = containerArray.getJSONObject(index);
+          addContainer(new Container(containerMetadata, this));
+        }
+        break;
+
+      default:
+        throw new IllegalArgumentException("Unsupported account metadata version " + accountMetadataVersion);
+    }
+  }
+
+  /**
+   * Constructor of Account without any {@link Container}.
+   * @param id The id of the account.
+   * @param name The name of the account.
+   * @param status The status of the account.
+   * @throws JSONException
+   */
+  public Account(short id, String name, String status) throws JSONException {
+    if (name == null) {
+      throw new IllegalArgumentException("name cannot be null.");
+    }
+    this.id = id;
+    this.name = name;
+    this.status = status;
+    metadata = new JSONObject();
+    metadata.put(ACCOUNT_METADATA_VERSION_KEY, ACCOUNT_METADATA_VERSION_1);
+    metadata.put(ACCOUNT_ID_KEY, id);
+    metadata.put(ACCOUNT_NAME_KEY, name);
+    metadata.put(ACCOUNT_STATUS_KEY, status);
+    metadata.put(CONTAINERS_KEY, new JSONArray());
+  }
+
+  /**
+   * Gets the id of the account.
+   * @return The id of the account.
+   */
+  public short getId() {
+    return id;
+  }
+
+  /**
+   * Gets the name of the account.
+   * @return The name of the account.
+   */
+  public String getName() {
+    return name;
+  }
+
+  /**
+   * Gets the metadata of the account in JSON format.
+   * @return The metadata of the account.
+   */
+  public JSONObject getMetadata() {
+    return metadata;
+  }
+
+  /**
+   * Gets the status of the account.
+   * @return The status of the account.
+   */
+  public String getStatus() {
+    return status;
+  }
+
+  /**
+   * Gets the {@link Container} of this account with the specified container id.
+   * @param containerId The id of the container to get.
+   * @return The {@link Container} of this account with the specified id, or {@code null} if such a
+   *                    container does not exist.
+   */
+  public Container getContainerByContainerId(short containerId) {
+    return containerIdToContainerMap.get(containerId);
+  }
+
+  /**
+   * Gets the {@link Container} of this account with the specified container name.
+   * @param containerName The name of the container to get.
+   * @return The {@link Container} of this account with the specified name, or {@code null} if such a
+   *                    container does not exist.
+   */
+  public Container getContainerByContainerName(String containerName) {
+    return containerNameToContainerMap.get(containerName);
+  }
+
+  /**
+   * Gets all the containers of this account in a list.
+   * @return All the containers of this account.
+   */
+  public List<Container> getAllContainers() {
```

Comment:
It seems in other places our code returns `List` as well, such as get `DataNodeId` and `PartitionId` in `ClusterMap`. I am inclined to returning a list.